models/user/user.go

```diff
@@ -729,6 +734,15 @@ func createUser(ctx context.Context, u *User, createdByAdmin bool, overwriteDefa
 	return committer.Commit()
 }
 
+func HitCreationLimit(ctx context.Context) bool {
+	// don't bother calling DB if limit not set
+	if setting.Service.MaxUserCreationLimit < 0 ||
+		int64(setting.Service.MaxUserCreationLimit) < CountUsers(ctx, &CountUserFilter{}) {
```

Comment:
Shouldn't it be a concurrent limit? Deactivated users should not be counted.

Reply:
However, doing this means we also need to keep the user count in mind when re-activating a user, and the `CountUsers` must ignore deactivated users.

Reply:
I think this feature should be similar to "you have 10 user licenses". Usually if someone leaves, you just deactivate the account and get 1 license back. This would not be possible with the current code.